frontend/public/components/instantiate-template.tsx

```diff
@@ -221,7 +256,8 @@ class TemplateForm_ extends React.Component<
     this.setState({ error: '', inProgress: true });
     this.createTemplateSecret()
       .then((secret: K8sResourceKind) => {
-        return this.createTemplateInstance(secret).then(() => {
+        return this.createTemplateInstance(secret).then((templateInstance: K8sResourceKind) => {
+          this.updatesecretOwnerRef(secret, templateInstance);
```

Comment:
Ideally we should not finish submission process until the owner ref is updated. Since this gives a promise, we could await on it to wait until completion.
```suggestion
          await this.updatesecretOwnerRef(secret, templateInstance);
```

Reply:
Thanks!
Have changed the codes accordingly